subsys/bluetooth/scan.c

```diff
@@ -220,7 +260,11 @@ static struct bt_scan {
 	 */
 	struct bt_le_conn_param conn_param;
 
+	/* Scan Connection attempts filter. */
```

Comment:
In order to save memory these should not be part of the struct if the feature has not been enabled.

You might have to switch to using `#if CONFIG_` instead of `if (IS_ENABLED(CONFIG_))` some places, but we should make sure to save as much memory as we can.